scipy/stats/_morestats.py

```diff
@@ -2068,16 +2067,10 @@ def shapiro(x):
     if N < 3:
         raise ValueError("Data must be at least length 3.")
 
-    a = zeros(N//2, dtype=np.float64)
-    init = 0
-
     y = sort(x)
     y -= x[N//2]  # subtract the median (or a nearby value); see gh-15777
 
-    w, pw, ifault = swilk(y, a, init)
-    if ifault not in [0, 2]:
-        warnings.warn("scipy.stats.shapiro: Input data has range zero. The"
-                      " results may not be accurate.", stacklevel=2)
```

Comment:
Given a sample with only one unique value, the implementation naturally produces NaN for the statistic and p-value.